documents/generic/model.go

```diff
@@ -0,0 +1,303 @@
+package generic
+
+import (
+	"encoding/json"
+	"reflect"
+
+	"github.com/centrifuge/centrifuge-protobufs/documenttypes"
+	"github.com/centrifuge/centrifuge-protobufs/gen/go/coredocument"
+	genericpb "github.com/centrifuge/centrifuge-protobufs/gen/go/generic"
+	"github.com/centrifuge/go-centrifuge/documents"
+	"github.com/centrifuge/go-centrifuge/errors"
+	"github.com/centrifuge/go-centrifuge/identity"
+	"github.com/centrifuge/precise-proofs/proofs"
+	"github.com/centrifuge/precise-proofs/proofs/proto"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/any"
+)
+
+const (
+	prefix string = "generic"
+
+	scheme = prefix
+)
+
+// tree prefixes for specific to documents use the second byte of a 4 byte slice by convention
+func compactPrefix() []byte { return []byte{0, 5, 0, 0} }
+
+// Data is a empty  structure.
+type Data struct{}
+
+// Generic implements the documents.Model for Generic documents
+type Generic struct {
+	*documents.CoreDocument
+	Data Data
```

Comment:
why do we need this at all?

Reply:
At the moment, we dont but we would need this soon.

Reply:
Hence, keeping it empty would